pyro/optim/optim.py

```diff
@@ -52,8 +66,10 @@ def __call__(self, params,  *args, **kwargs):
             if isinstance(self.optim_objs[p], torch.optim.lr_scheduler._LRScheduler) or \
                     isinstance(self.optim_objs[p], torch.optim.lr_scheduler.ReduceLROnPlateau):
                 # if optim object was a scheduler, perform an optimizer step
+                self.grad_clip[p](p)
```

Comment:
should we short circuit these calls if they are no-ops?

Reply:
I was trying to think of a good way to do that earlier and came up blank, but I will add one possible approach.